### PR TITLE
More improvements to the docs generation

### PR DESCRIPTION
**Pull request by tas50**
Keeping this open to add resource updates to over time

### CODE REVIEW

File: tasks/docs.rb [hunk 1]
```diff
@@ -305,7 +305,7 @@ def build_resource_data(name, data)
       else
         puts "Writing out #{resource}."
         FileUtils.mkdir_p "docs_site/#{resource}"
-        File.open("docs_site/#{resource}/_index.md", "w") { |f| f.write(resource_data.to_yaml) }
+        File.open("docs_site/#{resource}/_index.md", "w") { |f| f.write(resource_data.to_yaml + "---") }
```
Comment (phiggins) [inline tasks/docs.rb:308]: Nitpick, but a comment about why the `"---"` got added would be good so it doesn't get left off in the future.

File: lib/chef/resource/apt_preference.rb [hunk 2]
```diff
@@ -36,7 +36,7 @@ class AptPreference < Chef::Resource
         validation_message: "The provided package name is not valid. Package names can only contain alphanumeric characters as well as _, -, +, or *!"
 
       property :glob, String,
-        description: "Pin by glob() expression or with regular expressions surrounded by /."
+        description: "Pin by a `glob()` expression or with a regular expressions surrounded by `/`."
```
Comment (phiggins) [inline lib/chef/resource/apt_preference.rb:39]:
```suggestion
        description: "Pin by a `glob()` expression or with a regular expression surrounded by `/`."
```

File: tasks/docs.rb [hunk 3]
```diff
@@ -153,19 +153,41 @@ def friendly_default_value(property)
...
+      end
+
+      # add the special case for nothing
+      list[:nothing] = { "shortcode" => "resources_common_actions_nothing.md" }
+
+      list
+    end
+
     # TODO:
     # - what to do about "lazy default" for default?
```
Comment (phiggins) [inline tasks/docs.rb:177]:
```suggestion
```

Extremely nitpick, but I added this comment and then I think you addressed that a day or two later?

Comment (tas50) [reply tasks/docs.rb:177]: the lazy bit is still there and I keep seeing this and wondering what the right solution is? Maybe some warnings in the script or something.

Comment (phiggins) [reply tasks/docs.rb:177]: Ahh ok.

**Pull request merged**